homeassistant/components/tuya/fan.py

```diff
@@ -250,6 +250,8 @@ def percentage(self) -> int | None:
         if self._speeds is not None:
             if (value := self.device.status.get(self._speeds.dpcode)) is None:
                 return None
+            if value not in self._speeds.range:
```

Comment:
How can it public a speed not part of the range it set?
That sounds like an upstream bug.